### ISSUE TITLE
mocking a stream and whenCalling doesnt mock 

### ISSUE THREAD

**Issue by kazerdira** [atn832/fake_cloud_firestore#311]
i have this method 

 ResultStream<List<HelpResponse>> getResponses({required String requestId}) {
    return _remoteDataSource
        .getResponses(requestId)
        .map<EitherResult<List<HelpResponse>>>((response) {
      try {
        final helpResponses =
            response.map((model) => model as HelpResponse).toList();
        return Right(helpResponses);
      } catch (e) {
        return Left(
          ServerFailure(
            message: 'Error processing response: $e',
            statusCode: 1507,
          ),
        );
      }
    }).handleError((Object error) {
      if (error is ServerException) {
        return Left<Failure, dynamic>(
          ServerFailure(
            message: error.toString(),
            statusCode: error.statusCode ?? 501,
          ),
        );
      }
      return Left<Failure, dynamic>(
        ServerFailure(
          message: 'Unknown error occurred',
          statusCode: 500,
        ),
      );
    });
  }

and this is my test 

test('getRequests handles errors', () async {
      final collection = mockFirebaseFirestore.collection('responses');
        whenCalling(Invocation.method(#snapshots, null))
          .on(collection)
          .thenThrow(FirebaseException(
            plugin: 'cloud_firestore',
            message: 'Test error',
            code: 'test_error',
          ));

      // final stream = dataSource.getResponses(requestId);

      expect(
        () => dataSource.getResponses(requestId),
        emitsError(
          isA<ServerException>()
              .having((e) => e.message, 'message', contains('Test error'))
              .having((e) => e.statusCode, 'statusCode', 'test_error'),
        ),
      );
    });
  });
  
  but i am getting this error 
  
  
Expected: should emit an error that <Instance of 'ServerException'> with `message`: contains 'Test error' and `statusCode`: 'test_error'
  Actual: <Closure: () => Stream<List<HelpResponseModel>>>
   Which: was not a Stream or a StreamQueue

**Comment by atn832** [comment]
You are right, throwing errors is not supported on collection.snapshots. See the list of supported methods here: https://github.com/atn832/fake_cloud_firestore?tab=readme-ov-file#mocking-exceptions. If you need exceptions thrown when calling snapshots, feel free to propose a PR and I'll review it!
